### PR TITLE
Issue#21

### PR DESCRIPTION
**Pull request by GrzegorzDryja**
Check [last commit](https://github.com/tomkowy/notatnik_parser/commit/185cef0dd94c6b86f99cefc0bf457a4e0d990e5a).

### CODE REVIEW

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 1]
```diff
@@ -5,35 +5,141 @@
...
+
+            try
+            {
+                using (FileStream fs = File.Create(pathAndFileName))
+                {
+                    Byte[] textNote = new UTF8Encoding(true).GetBytes(textNoteParameter);
+                    fs.Write(textNote, 0, textNote.Length);
+                }
+            }
+            catch (UnauthorizedAccessException)
```
Comment (tomku-mar) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs]: why you catch this type of exception

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 2]
```diff
@@ -5,35 +5,141 @@
...
+            }
+        }
+
+        private void WriteNewNote()
+        {       
+            Console.WriteLine("Wprowadź notatkę (wciśnij 2 x <enter> aby zapisać):");
+            string note = Console.ReadLine();
+            ConsoleKeyInfo typedKey = Console.ReadKey();
+
+            if (typedKey.Key == ConsoleKey.Enter && SaveFile(note))
```
Comment (tomku-mar) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs]: saveFile shoudn't execute in if. You should execude saveFile method and then pass returned value to if statement. You can try at first ask about new path and check it and then ask about data for file. Now save method method saves the file and asks about path. It's two responsibilities

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 3]
```diff
@@ -5,35 +5,141 @@
...
+                FileAttributes attr = (new FileInfo(pathAndFileName)).Attributes;
+                return false;
+            }
+            
+            return true;
+        }
+
+        private void OpenTextFile(string insertPathValue)
+        {
+            using (FileStream fs = File.Open(insertPathValue, FileMode.Open))
```
Comment (tomkowy) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs]: You don't check that file exist. Maybe someone deleted it.

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 4]
```diff
@@ -5,35 +5,141 @@
...
+                UTF8Encoding temp = new UTF8Encoding(true);
+
+                while (fs.Read(b,0,b.Length) > 0) 
+                {
+                    Console.WriteLine(temp.GetString(b));
+                }
+            }
+        }
+
+        private bool IsUserPathInputValid(string insertPathValue)
```
Comment (tomku-mar) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs]: This method check input and open file. You should create a method to check input and open fil(this one you have :)) and use their separately in MenageUserPathInput

**Review by tomku-mar — CHANGES_REQUESTED** [review]
it's very good. You started to separate responsibility

**Comment by GrzegorzDryja** [comment]
Thank you for your review @tomkowy, I really appreciate your help which is invaluable. I will include your suggestions in next commit.

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs (hunk 1 above)
Comment (GrzegorzDryja) [reply Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs]: The story is simple... I got this error in IDE during tests, I mentioned that I thoroughly tested this code. Sometimes happens that you try to save something somewhere where you aren't allowed to save this.

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 5]
```diff
@@ -5,35 +5,207 @@
...
+        }
+
+        private bool IsUserInputValid(string userInput)
+        {
+            if (userInput == "1" || userInput == "2")
+            {
+                return true;
+            }
+
+            Console.WriteLine("Proszę wybrać numer z listy");
```
Comment (tomku-mar) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:43]: move this somewere else. You have two dependencies: check input and show messages.

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 6]
```diff
@@ -5,35 +5,207 @@
...
+        private string MenageUserNameInput(string pathName)
+        {
+            string insertNameValue;
+            string insertPathAndNameValue;
+
+            do
+            {   
+                Console.WriteLine("Podaj nazwę pliku z rozszerzeniem:");                
+                insertNameValue = Console.ReadLine();
+                insertPathAndNameValue = pathName + '/' + insertNameValue;
```
Comment (tomku-mar) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:139]: check method String.Format and better, string interpolation ($"")

File: Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs [hunk 7]
```diff
@@ -5,35 +5,207 @@
...
+            while (!IsFileExist(insertPathAndNameValue));
+
+            return insertPathAndNameValue;
+        }
+
+        private bool IsFileExist(string insertPathAndNameValue)
+        {
+            if(File.Exists(insertPathAndNameValue))
+            {                
+                Console.WriteLine("{0} - ta ścieżka jest niewłaściwa.", insertPathAndNameValue);
```
Comment (tomku-mar) [inline Notatnik_parser/Notatnik_parser/Notepad/NotepadMenu.cs:150]: again, two dependencies. The methods's name tells us that this is a file check

**Review by tomku-mar — APPROVED** [review]
In general is good. I add some comments, but you don't need commit it :)

**Pull request merged**